### PR TITLE
[SUPPORT][LIVE-10228] add evm-family-integration-process doc

### PR DESCRIPTION
**Pull request by chabroA**
### 📝 Description

Document EVM family integration process

### ❓ Context

- https://ledgerhq.atlassian.net/browse/LIVE-10228

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] ~`npx changeset` was attached.~ N/A
- [ ] ~**Covered by automatic tests.**~ N/A 
- [ ] ~**Impact of the changes:**~ N/A 
 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Most EVM coin integrations are quite similar. The main specs will be outlined below.
+But each coin may have its specificities and more information can be found on the specific coin integration Jira Epic created for each integration, based on this template https://ledgerhq.atlassian.net/browse/LIVE-10187
+
+- Integration is done in Ledger Live desktop and Ledger Live mobile
+- Supported features:
+  - Sending and receiving coins
+  - Viewing transaction history
+  - Token support, enabling interactions with the blockchain tokens
+- Use of public RPC node and explorer (etherscan or blockscout-like). This should be specified in the corresponding Jira Epic
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:18]:
```suggestion
- Use of public RPC node and explorer (etherscan-like or blockscout-like). This should be specified in the corresponding Jira Epic
```

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  - To find a public node, you can use https://chainlist.org/ for example
+  - To find a public explorer, you can use https://blockscan.com/ (etherscan-like) or https://www.blockscout.com/
+- The [Ethereum embedded app](https://github.com/LedgerHQ/app-ethereum) should handle this new network
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:21]:
```suggestion
- The [Ethereum embedded app](https://github.com/LedgerHQ/app-ethereum) must handle this new network
```

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+5. Add a new feature flag config for this currency:
+   1. The new feature flag type in the [`CurrencyFeatures`](https://github.com/LedgerHQ/ledger-live/blob/7513354754ce0326a4ebcbcd86a5b4b38898a49e/libs/ledgerjs/packages/types-live/src/feature.ts#L80-L124) type under [`libs/ledgerjs/packages/types-live/src/feature.ts`](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/types-live/src/feature.ts)
+   2. The new feature flag definition with default value in the [`CURRENCY_DEFAULT_FEATURES`](https://github.com/LedgerHQ/ledger-live/blob/7513354754ce0326a4ebcbcd86a5b4b38898a49e/libs/ledger-live-common/src/featureFlags/defaultFeatures.ts#L23-L67) mapping under [`libs/ledger-live-common/src/featureFlags/defaultFeatures.ts`](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/featureFlags/defaultFeatures.ts)
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:37]: Will just need to update the file path when https://github.com/LedgerHQ/ledger-live/pull/5341 is merged

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+6. Update related doc and snapshot files:
+   1. run `pnpm doc` under `libs/ledgerjs/packages/types-live` to update the doc
+   2. run `pnpm test:jest` under `apps/ledger-live-desktop` to update the snapshots
+7. Add the related currency icon as SVG format under the [`libs/ui/packages/crypto-icons/src/svg`](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ui/packages/crypto-icons/src/svg) folder within the [@ledgerhq/icons-ui](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ui/packages/crypto-icons) package
+   1. the file name should follow the `{currency_id}.svg` naming convention
+   2. the SVG icon should follow Ledger Live currency icon guideline (TBD)
+
+### Optional steps
+
+_Optionnal / extra steps that might be needed on a case-by-case basis depending on the integration_
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:48]:
```suggestion
_Optional / extra steps that might be needed on a case-by-case basis depending on the integration_
```

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+- If the related currency public explorer is not an etherscan or blockscout-like one, you might need to add a new implementation for this currency explorer
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:50]:
```suggestion
- If the related currency public explorer is not an etherscan-like or blockscout-like one, you might need to add a new implementation for this currency explorer
```

File: libs/coin-evm/docs/evm-family-integration-process/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+   3. Use this new feature flag in both [LLD](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/modals/AddAccounts/steps/StepChooseCurrency.tsx) and [LLM](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-mobile/src/screens/AddAccounts/01-SelectCrypto.tsx) currencies entry points (add account / select crypto flow)
+6. Update related doc and snapshot files:
+   1. run `pnpm doc` under `libs/ledgerjs/packages/types-live` to update the doc
+   2. run `pnpm test:jest` under `apps/ledger-live-desktop` to update the snapshots
+7. Add the related currency icon as SVG format under the [`libs/ui/packages/crypto-icons/src/svg`](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ui/packages/crypto-icons/src/svg) folder within the [@ledgerhq/icons-ui](https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ui/packages/crypto-icons) package
+   1. the file name should follow the `{currency_id}.svg` naming convention
+   2. the SVG icon should follow Ledger Live currency icon guideline (TBD)
```
Comment (haammar-ledger) [inline libs/coin-evm/docs/evm-family-integration-process/README.md:44]: You can mention this tool, that's available to anyone and can be used to validate icon SVGs for LL.
https://live.ledger.tools/svg-icons

**Review by haammar-ledger — COMMENTED** [review]
:fire:

File: libs/coin-evm/docs/evm-family-integration-process/README.md (hunk 3 above)
Comment (chabroA) [reply libs/coin-evm/docs/evm-family-integration-process/README.md:37]: Indeed 👍 
This doc (as it's the case for any doc) should be regularly updated to closely fit the current state of the codebase

File: libs/coin-evm/docs/evm-family-integration-process/README.md (hunk 6 above)
Comment (chabroA) [reply libs/coin-evm/docs/evm-family-integration-process/README.md:44]: Wasn't even aware of this tool's existence 😮 

Will update the doc to mention it 👍

Comment (chabroA) [reply libs/coin-evm/docs/evm-family-integration-process/README.md]: Done here 6572c4f6d6ac5cb5ba7f9a361e1b24198f03ddb0

**Review by hedi-edelbloute — APPROVED** [review]
LGTM

**Comment by chabroA** [comment]
> As a general comment, you can say "etherscan-like" and never mention "blockscout-like". Blockscout is either a white label product or an open-source self-hosted solution which is compatible with our stack specifically because it's already an "etherscan-like" solution by having a compatibility layer with the Etherscan API. Blockscout explorers use the "etherscan" adapters.
> 
> The mention of blockscout in our stack is technically useless today, it's only here so we have the flexibility later on to make each implementation have a different behaviour if needed (for example, blockscout explorers are generally not rate limited, so a specific behaviour could be coded based on that).

Updated here f8f244f37260980b9f4e2a4429b469873d6fb769 👍

**Pull request merged**